src/OrchardCore.Modules/OrchardCore.Microsoft.Authentication/Configuration/MicrosoftAccountOptionsConfiguration.cs

```diff
@@ -54,22 +60,29 @@ public void Configure(AuthenticationOptions options)
         public void Configure(string name, MicrosoftAccountOptions options)
         {
             // Ignore OpenID Connect client handler instances that don't correspond to the instance managed by the OpenID module.
-            if (!string.Equals(name, MicrosoftAccountDefaults.AuthenticationScheme))
+            if (!String.Equals(name, MicrosoftAccountDefaults.AuthenticationScheme))
             {
                 return;
             }
 
             var loginSettings = GetMicrosoftAccountSettingsAsync().GetAwaiter().GetResult();
+            if (loginSettings == null)
+            {
+                return;
+            }
 
-            options.ClientId = loginSettings?.AppId ?? string.Empty;
+            options.ClientId = loginSettings.AppId ?? String.Empty;
```

Comment:
Makes sense here

Reply:
But not necessary, because as for other, here `GetMicrosoftAccountSettingsAsync()` call `ValidateSettings()` that checks `AppId`, and return null if the validation fails, which is checked just before line 69